go-controller/pkg/ovn/controller/services/loadbalancer.go

```diff
@@ -136,13 +218,23 @@ func EnsureLBs(nbClient libovsdbclient.Client, service *corev1.Service, existing
 		mapLBDifferenceByKey(removeLBsFromGroups, existingGroups, wantGroups, blb)
 	}
 
-	ops, err := libovsdbops.CreateOrUpdateLoadBalancersOps(nbClient, nil, existinglbs...)
+	ops, err := libovsdbops.CreateOrUpdateLoadBalancersOps(nbClient, nil, toNBLoadBalancerList(existinglbs)...)
+	if err != nil {
+		return err
+	}
+
+	ops, err = libovsdbops.CreateLoadBalancersOps(nbClient, ops, toNBLoadBalancerList(newlbs)...)
 	if err != nil {
 		return fmt.Errorf("failed to create ops for ensuring update of service %s/%s load balancers: %w",
 			service.Namespace, service.Name, err)
 	}
 
-	ops, err = libovsdbops.CreateLoadBalancersOps(nbClient, ops, newlbs...)
+	ops, err = libovsdbops.CreateChassisTemplateVarOps(nbClient, ops, toNBTemplateList(newlbs)...)
+	if err != nil {
+		return err
+	}
+
+	ops, err = libovsdbops.CreateOrUpdateChassisTemplateVarOps(nbClient, ops, toNBTemplateList(existinglbs)...)
```

Comment:
So the reason we have LB ops split in `CreateOrUpdateLoadBalancersOps` and `CreateLoadBalancersOps` is to avoid running the LB name matching predicate for LB rows that we know don't exist.

But I don't think we need that kind of split for TemplateVar. On one side, the existence of a chassis in TemplateVar table is not tied to the existence of LBs. On the other side,  the existence of a chassis in TemplateVar table is most likely true at this point other than exceptional cases which I will talk about.

So just having a `CreateOrUpdateChassisTemplateVarOps` might be enough.

Reply:
But now to the point of whether a chassis row in TemplateVar table exists or not. I am assuming this row is not added by OVN.

The row will most likely be added in the service controller `RequestFullSync` or here. The row is deleted if needed in the default network controller. These two run off from a different informer factory. So a race condition where a node is removed, the row deleted but then added back by the service controller that was synchronizing at the same time is possible.

So perhaps the row should be added by the default network controller as well. And here just be strictly updated. If the row does not exist here we can fail which will re-queue the service synchronization for a later time.

Reply:
I see there is a periodic chassis cleanup going on. So maybe we don't mind having a stale chassis TemplateVar row for a while and at the same time it might be sub-optimal to fail a sync just because of this. 

I would still add the consideration that splitting the responsibility of adding the row and removing it to different components does not look great.

Reply:
Another problem is that `RequestFullSync` failures don't do anything; they don't trigger a retry.  So if `RequestFullSync` fails to create the Chassis_Template_Var we kind of have a problem.

So, it seems to me we need to create-or-update everywhere:
a. in the default network controller (just an empty record per node)
b. in `RequestFullSync()`, setting the node IP template values.
c. in `EnsureLBs()`, setting the load balancer backend template values.

Should we also panic on error in `RequestFullSync()`?

And I think what i'm doing is slightly wrong:  I can't resync node IP templates only in `RequestFullSync()` because the chassisID might not be set yet there.

Reply:
> Another problem is that `RequestFullSync` failures don't do anything; they don't trigger a retry. So if `RequestFullSync` fails to create the Chassis_Template_Var we kind of have a problem.
> 
> So, it seems to me we need to create-or-update everywhere: a. in the default network controller (just an empty record per node) b. in `RequestFullSync()`, setting the node IP template values. c. in `EnsureLBs()`, setting the load balancer backend template values.

While trying to implement this I realized we don't have the chassis-id in the default network controller when we create the logical network for new nodes.  ovnkube-node adds the chassis-id to the Node annotation.  So, I decided to keep it as it was: `RequestFullSync()` will create all required node IP template records if they don't exist and will update existing ones.